tests/posit/connect/test_permissions.py

```diff
@@ -0,0 +1,192 @@
+    @responses.activate
+    def test_request_shape(self):
+        # test data
+        guid = str(uuid.uuid4())
+        content_guid = str(uuid.uuid4())
+        principal_guid = str(uuid.uuid4())
+        principal_type = "principal_type"
+        role = "role"
+        extraneous = "extraneous"
+
+        # define api behavior
+        responses.put(
+            f"https://connect.example/__api__/v1/content/{content_guid}/permissions/{guid}",
+            json={
+                # doesn't matter for this test
+            },
+            match=[
+                # assertion
+                matchers.json_params_matcher(
+                    {
+                        # validate that initial permission fields are set
+                        "principal_guid": principal_guid,
+                        "principal_type": principal_type,
+                        "role": role,
+                        # validate that arguments passed to update are set
+                        "extraneous": extraneous,
+                    }
+                )
+            ],
+        )
+
+        # setup
+        config = Config(api_key="12345", url="https://connect.example/")
+        session = requests.Session()
+        permission = Permission(
+            config,
+            session,
+            guid=guid,
+            content_guid=content_guid,
+            principal_guid=principal_guid,
+            principal_type=principal_type,
+            role=role,
+        )
+
+        # invoke
+        # assertion occurs in match above
+        permission.update(extraneous=extraneous)
+
+    @responses.activate
+    def test_role_update(self):
+        # test data
+        old_role = "old_role"
+        new_role = "new_role"
+
+        # define api behavior
+        guid = str(uuid.uuid4())
+        content_guid = str(uuid.uuid4())
+        responses.put(
+            f"https://connect.example/__api__/v1/content/{content_guid}/permissions/{guid}",
+            json={"role": new_role},
+            match=[
+                matchers.json_params_matcher(
+                    {
+                        "principal_guid": None,
+                        "principal_type": None,
+                        "role": new_role,
+                    }
+                )
+            ],
+        )
+
+        # setup
+        config = Config(api_key="12345", url="https://connect.example/")
+        session = requests.Session()
+        permission = Permission(
+            config, session, guid=guid, content_guid=content_guid, role=old_role
+        )
+
+        # assert role change with respect to api response
+        assert permission.role == old_role
+        permission.update(role=new_role)
+        assert permission.role == new_role
+
+
+class TestPermissionsFind:
+    @responses.activate
+    def test(self):
+        # test data
+        content_guid = str(uuid.uuid4())
+        fake_permissions = [
+            {
+                "guid": str(uuid.uuid4()),
```

Comment:
But isn't the field called "id" in the API?

Also, can we put these JSON API responses in a file like the others?